File: Application/src/main/java/com/shareyourproxy/api/rx/command/eventcallback/LoggedInUserUpdatedEventCallback.java
package com.shareyourproxy.api.rx.command.eventcallback;

import android.os.Parcel;
import android.support.annotation.NonNull;

import com.shareyourproxy.api.domain.model.User;

/**
 * Created by Evan on 6/9/15.
 */
public class LoggedInUserUpdatedEventCallback extends UserEventCallback {
    private final static java.lang.ClassLoader CL =
        LoggedInUserUpdatedEventCallback.class.getClassLoader();
    public static final Creator<LoggedInUserUpdatedEventCallback> CREATOR =
        new Creator<LoggedInUserUpdatedEventCallback>() {
            @Override
            public LoggedInUserUpdatedEventCallback createFromParcel(Parcel in) {
                return new LoggedInUserUpdatedEventCallback(
                    (User) in.readValue(CL));
            }

            @Override
            public LoggedInUserUpdatedEventCallback[] newArray(int size) {
                return new LoggedInUserUpdatedEventCallback[size];
            }
        };
    public LoggedInUserUpdatedEventCallback(@NonNull User user) {
        super(user);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/command/eventcallback/LoggedInUserUpdatedEventCallback.kt
package com.shareyourproxy.api.rx.command.eventcallback

import android.os.Parcel
import android.os.Parcelable

import com.shareyourproxy.api.domain.model.User

/**
 * Current user updated.
 */
internal final class LoggedInUserUpdatedEventCallback(user: User) : UserEventCallback(user) {
    companion object {
        private val CL = LoggedInUserUpdatedEventCallback::class.java.classLoader
        @JvmField val CREATOR: Parcelable.Creator<LoggedInUserUpdatedEventCallback> = object : Parcelable.Creator<LoggedInUserUpdatedEventCallback> {

            override fun createFromParcel(parcel: Parcel): LoggedInUserUpdatedEventCallback {
                return LoggedInUserUpdatedEventCallback(parcel.readValue(CL) as User)
            }

            override fun newArray(size: Int): Array<LoggedInUserUpdatedEventCallback?> {
                return arrayOfNulls(size)
            }
        }
    }
}
